### ISSUE TITLE
geckodriver never returns when action sequence causes navigation

### ISSUE THREAD

**Issue by pbi-qfs** [mozilla/geckodriver#1063]
## System
* Version: 0.19.1
* Platform: Windows, Mac
* Firefox: Since 57.0
* Selenium: Java Bindings, since 3.6.0

## Testcase

```js
driver.get("https://typo3.org/typo3-cms/");
WebDriverWait wait = new WebDriverWait(driver, 10);

for (int i = 0; i < 10; i++) {
    final WebElement header = wait
            .until(ExpectedConditions.elementToBeClickable(By.xpath("/html/body/div[1]/div[2]/ul/li[3]/a")));
    final Actions builder = new Actions(driver);
    builder.moveToElement(header, 0, 0);
    builder.click();
    builder.moveByOffset(0, 0);
    builder.perform();
}
```

## Trace-level log

```
1510842474040	webdriver::server	DEBUG	-> POST /session/d83dd16f-a20e-1144-b13d-d63fbfa3d8c3/actions {"actions":[{"id":"default mouse","type":"pointer","parameters":{"pointerType":"mouse"},"actions":[{"duration":100,"x":0,"y":0,"type":"pointerMove","origin":{"ELEMENT":"c810203c-06b6-3f47-b39a-fe1d4776c106","element-6066-11e4-a52e-4f735466cecf":"c810203c-06b6-3f47-b39a-fe1d4776c106"}},{"button":0,"type":"pointerDown"},{"button":0,"type":"pointerUp"},{"duration":200,"x":0,"y":0,"type":"pointerMove","origin":"pointer"}]}]}
1510842474041	geckodriver::marionette	TRACE	-> 398:[0,6,"performActions",{"actions":[{"actions":[{"duration":100,"origin":{"element-6066-11e4-a52e-4f735466cecf":"c810203c-06b6-3f47-b39a-fe1d4776c106"},"type":"pointerMove","x":0,"y":0},{"button":0,"type":"pointerDown"},{"button":0,"type":"pointerUp"},{"duration":200,"origin":"pointer","type":"pointerMove","x":0,"y":0}],"id":"default mouse","parameters":{"pointerType":"mouse"},"type":"pointer"}]}]
1510842474042	Marionette	TRACE	0 -> [0,6,"performActions",{"actions":[{"actions":[{"duration":100,"origin":{"element-6066-11e4-a52e-4f735466cecf":"c810203c-06b6-3f47-b39a-fe1d4776c106"},"type":"pointerMove","x":0,"y":0},{"button":0,"type":"pointerDown"},{"button":0,"type":"pointerUp"},{"duration":200,"origin":"pointer","type":"pointerMove","x":0,"y":0}],"id":"default mouse","parameters":{"pointerType":"mouse"},"type":"pointer"}]}]
```

## Expected output

Should click the link 10 times, but stalls before.

**Comment by whimboo** [comment]
@mjzffr could you have a look at this issue?

**Comment by codingphil** [comment]
Can observe the same behavior in my nightly regression test. If repeated actions calls are happening the /actions call to the driver does never return.

In my environment this is:
1. navigate
2. click with actions like 
`new Actions(driver).moveToElement(signUpButton).click().perform();`
This one freezes.
3. verify
Repeat with next similar test.

**Comment by andreastt** [comment]
Filed https://bugzilla.mozilla.org/show_bug.cgi?id=1431116.

**Comment by mjzffr** [comment]
Hmm, I guess we only partly fixed this in https://bugzilla.mozilla.org/show_bug.cgi?id=1345653
There's a corresponding wdspec test and all, but it seems not to cover this scenario.

**Comment by pchtest** [comment]
I observed the same behavior in my testing today with the upgrade of slenium, Firefox.
OS: windows7.0
Selenium: 3.8.1
Firefox: 56.0.2 (Tried 57.0.3,still the same)
gecho driver: 0.19.0
Java: 8
in my Selenium /Java scripts, no issues with Java 8/FF47/selenium2.58
I have upgdared all to the above and click on login page is not responding.

here is my code:

if (sBrowser.equals("*firefox")){
	         		System.setProperty("webdriver.gecko.driver","C://SeleniumJavaClient//geckodriver//geckodriver.exe");
	         		driver = new FirefoxDriver();
				    driver.manage().window().maximize();
					driver.manage().timeouts().implicitlyWait(10, TimeUnit.SECONDS);

public static void Login(String username, String password) throws Exception{
		try {
			if(driver.findElements(By.xpath("//div[text()='Error']")).size()>0){
				driver.findElement(By.xpath("//button[@class='buttonOk']")).click();
			}
			driver.findElement(By.xpath(loginObject.getProperty("username"))).clear();
			Thread.sleep(2000);
			driver.findElement(By.xpath(loginObject.getProperty("password1"))).clear();
			Thread.sleep(2000);
			driver.findElement(By.xpath(loginObject.getProperty("username"))).sendKeys(username);
			driver.findElement(By.xpath(loginObject.getProperty("password1"))).sendKeys(password);
			driver.findElement(By.xpath(loginObject.getProperty("signin_button"))).click();
			Thread.sleep(100000);

Expected:  signin button click should be success and navigate to the home page. But action is not responding.

Logs: 

[TestNG] Running:
  C:\Users\pchalla\AppData\Local\Temp\testng-eclipse-993991027\testng-customsuite.xml

1516227526929	geckodriver	INFO	geckodriver 0.19.0
1516227526940	geckodriver	INFO	Listening on 127.0.0.1:14971
1516227527710	mozrunner::runner	INFO	Running command: "C:\\Program Files\\Mozilla Firefox\\firefox.exe" "-marionette" "-profile" "C:\\Users\\pchalla\\AppData\\Local\\Temp\\rust_mozprofile.Vll3BbE6wIDJ"
1516227528693	Marionette	INFO	Enabled via --marionette
Unable to read VR Path Registry from C:\Users\pchalla\AppData\Local\openvr\openvrpaths.vrpath
[Parent 6272, Gecko_IOThread] WARNING: pipe error: 109: file z:/build/build/src/ipc/chromium/src/chrome/common/ipc_channel_win.cc, line 346
Unable to read VR Path Registry from C:\Users\pchalla\AppData\Local\openvr\openvrpaths.vrpath
[Child 6064, Chrome_ChildThread] WARNING: pipe error: 109: file z:/build/build/src/ipc/chromium/src/chrome/common/ipc_channel_win.cc, line 346
[Child 6064, Chrome_ChildThread] WARNING: pipe error: 109: file z:/build/build/src/ipc/chromium/src/chrome/common/ipc_channel_win.cc, line 346
1516227532734	Marionette	INFO	Listening on port 55328
1516227533180	Marionette	WARN	TLS certificate errors will be ignored for this session
1516227533257	Marionette	DEBUG	Register listener.js for window 4294967297
Jan 17, 2018 5:18:53 PM org.openqa.selenium.remote.ProtocolHandshake createSession
INFO: Detected dialect: W3C
1516227538696	Marionette	DEBUG	Received DOM event "beforeunload" for "about:blank"
1516227539363	Marionette	DEBUG	Received DOM event "pagehide" for "about:blank"
1516227539363	Marionette	DEBUG	Received DOM event "unload" for "about:blank"
1516227543070	Marionette	DEBUG	Received DOM event "DOMContentLoaded" for "https://myvoice-preprod.primus.ca/HPBX/"
1516227546884	Marionette	DEBUG	Received DOM event "pageshow" for "https://myvoice-preprod.primus.ca/HPBX/"
1516227571515	Marionette	DEBUG	Canceled page load listener because no navigation has been detected
